src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -430,6 +437,40 @@ void DynamoViewModelRequestAboutWindow(DynamoViewModel model)
             aboutWindow.ShowDialog();
         }
 
+        private void OutsideGalleryView_MouseLeftButtonDown(object sender, MouseButtonEventArgs e)
+        {
+            dynamoViewModel.CloseGalleryCommand.Execute(null);
+            e.Handled = true;
+        }
+
+        void DynamoViewModelRequestShowGallery()
+        {
+            if (galleryView == null) //On-demand instantiation
+            {
+                galleryView = new GalleryView(new GalleryViewModel(dynamoViewModel));
+                Grid.SetColumnSpan(galleryUi, mainGrid.ColumnDefinitions.Count);
+                Grid.SetRowSpan(galleryUi, mainGrid.RowDefinitions.Count);
+                galleryUi.Visibility = Visibility.Hidden;
+                galleryUi.Background = new SolidColorBrush(Colors.Black)
+                {
+                    Opacity = 0.8
+                };
+            }
+
+            if (GalleryViewModel.IsAnyContent)
+            {
+                galleryUi.Children.Add(galleryView);
+                galleryUi.Visibility = Visibility.Visible;
+                galleryView.Focus(); //get keyboard focus (for ESC)
+            }
+        }
+
+        void DynamoViewModelRequestCloseGallery()
+        {
+            galleryUi.Children.Remove(galleryView);
+            galleryUi.Visibility = Visibility.Hidden;
```

Comment:
Yes, it is. But I supposed you cannot requestclosegallery if was not added first?